src/main/groovy/com/github/dispader/manifesto/ManifestoPlugin.groovy

```diff
@@ -10,67 +10,23 @@ class ManifestoPlugin implements Plugin<Project> {
 
         project.extensions.create('manifesto', ManifestoPluginExtension, project)
 
+        def version = new Version()
+        project.version = version.version
+
         project.plugins.whenPluginAdded { plugin ->
             project.tasks.withType(Jar).each {
                 it.manifest.with {
-
-                    def version = new Version()
-                    if ( !version ) {
-                        project.logger.warn "warning: ${Version.warningText}"
-                        return
-                    }
-
-                    if ( !project.version || project.version == 'unspecified' ) {
-                        project.version = version.version
-                    } else {
-                        project.logger.warn "warning: Manifest version not set."
-                    }
-
-                    attributes('Specification-Title': project.name)
-
-                    if ( project.manifesto?.vendor ) {
```

Comment:
`project.manifesto?` here was to guard against when folks were using the default configuration (not adding any kind of `manifesto` config in their build, and just relying on the defaults.  I... think it still serves that purpose?  If we don't NPE on the new code starting around line 21 when there's no `manifesto { }` in a build, I'm way good losing it there.

Reply:
Oh, I'm sorry— I'm seeing the source of confusion now.  (Now that I read the commit comment carefully, which I should have done in the first place.)

The `project.manifesto?` reference here doesn't refer to the plugin at all— it's the configuration object that we create above to allow folks to configure things about the plugin in their build (`manifesto { }` in their build files).  We might want to lose this, but we're trying to null-safe dereferencing the Plugin itself, but the configuration object that may or may not have gotten added to the project.